### PR TITLE
Add DNS Hostname support, explicit Gateway Endpoints, and VPC Flow Logs

### PR DESCRIPTION
**Pull request by jricks92**
- Enables DNS Hostname support on the VPC.
- Explicitly configures S3 and DynamoDB endpoints as Gateway Endpoints to avoid confusion.
- Adds option to enable VPC Flow Logs that are delivered to CloudWatch Logs.

### CODE REVIEW

**Comment by dougireton** [comment]
@jricks92 this looks great! Can you update the VPC Template Parameters table in the README?

**Comment by jricks92** [comment]
> @jricks92 this looks great! Can you update the VPC Template Parameters table in the README?

Yep will do!

**Comment by jricks92** [comment]
@dougireton @stephanielingwood README and example parameters have been updated.

File: README.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ You are free to use any /16 to /28 CIDR block in the RFC 1918 private address ra
 
 ## VPC Template Parameters
 
-To deploy this VPC template, you'll need to know the VPC CIDR block, the three public, and three private subnet CIDR blocks. You will also choose whether the VPC will support highly available NAT Gateways, or, by default, a more cost effective single NAT Gateway.
+To deploy this VPC template, you'll need to know the VPC CIDR block, the three public, and three private subnet CIDR blocks. You will also choose whether the VPC will support highly available NAT Gateways and VPC Flow Logs, or, by default, a more cost effective single NAT Gateway without VPC Flow Logs enabled.
```
Comment (stephanielingwood) [inline README.md]: Can we edit this paragraph to more clearly indicate that the choices of highly available NAT and the choice to include flow logs are independent choices from one other? I can see where, reading this, someone could assume that the highly available NAT choice also controls the flow logs choice. (The table below helps, to be sure)

**Review by stephanielingwood — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (jricks92) [reply README.md]: Just updated in the latest commit

**Review by stephanielingwood — APPROVED** [review]
Thanks for all your work on this! Looks good to me :)

File: parameters/us-west-2/dev.json [hunk 2]
```diff
@@ -6,5 +6,6 @@
     "PublicAZBSubnetBlock=10.0.96.0/20",
     "PrivateAZCSubnetBlock=10.0.128.0/19",
     "PublicAZCSubnetBlock=10.0.160.0/20",
-    "HighlyAvailableNat=true"
+    "HighlyAvailableNat=true",
+    "EnableVpcFlowLogs=true"
```
Comment (dougireton) [inline parameters/us-west-2/dev.json:10]: So, why is `EnableVpcFlowLogs=true` here but `false` in `us-east-1` above?

**Review by dougireton — CHANGES_REQUESTED** [review]

File: parameters/us-west-2/dev.json (hunk 2 above)
Comment (jricks92) [reply parameters/us-west-2/dev.json:10]: Just to show an example that you could do either. Same reason why HA nat is True in us-west-2 but false in us-east-1. I was just following the existing pattern :)

**Pull request merged**